src/main/kotlin/org/jetbrains/research/tasktracker/tracking/TaskFileHandler.kt

```diff
@@ -55,27 +62,42 @@ object TaskFileHandler {
         }
     }
 
-    private fun removeVirtualFileListener(virtualFile: VirtualFile) {
-        ApplicationManager.getApplication().invokeAndWait {
-            val document = FileDocumentManager.getInstance().getDocument(virtualFile)
-            document?.removeDocumentListener(listener)
+    private fun removeVirtualFileListener(virtualFiles: List<VirtualFile>) {
+        virtualFiles.forEach { file ->
+            ApplicationManager.getApplication().invokeAndWait {
+                val document = FileDocumentManager.getInstance().getDocument(file)
+                document?.removeDocumentListener(listener)
+            }
         }
     }
 
-    private fun getOrCreateFile(project: Project, task: Task): VirtualFile? {
-        val relativeFilePath =
-            "${DefaultContentProvider.getDefaultFolderRelativePath(task)}/${task.relativeFilePath ?: ""}"
-        ApplicationManager.getApplication().runWriteAction {
-            addSourceFolder(relativeFilePath, ModuleManager.getInstance(project).modules.last())
-        }
-        val file = File("${project.basePath}/$relativeFilePath/${task.filename}${task.extension.ext}")
-        if (!file.exists()) {
+    private fun getOrCreateFiles(project: Project, task: Task): List<VirtualFile?> {
+        return task.taskFiles.map { taskFile ->
+            val relativeFilePath = taskFile.relativePath
             ApplicationManager.getApplication().runWriteAction {
-                FileUtil.createIfDoesntExist(file)
-                file.writeText(task.content ?: DefaultContentProvider.getDefaultContent(task))
+                addSourceFolder(relativeFilePath, ModuleManager.getInstance(project).modules.last())
             }
+            val path = getPath(project, taskFile, task)
+            val file = File(path)
+            if (!file.exists()) {
+                ApplicationManager.getApplication().runWriteAction {
+                    FileUtil.createIfDoesntExist(file)
```

Comment:
You don't need to check that the file does not exist, since this branch is only about the case when the file does not exist